### PR TITLE
Contrail v4. Container based deployment support.

### PR DESCRIPTION
**Pull request by alexey-mr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,86 +1,44 @@
...
------
-
-**What the puppet-contrail module affects**
-
-* puppet-contrail, the Juniper SDN service.
-
-### Installing puppet-contrail
-
-    example% puppet module install enovance/contrail
+The puppet-contrail module is a thorough attempt to make Puppet capable of managing the entirety of OpenContrail. The current implementation is limited to support Contrail v4.0 and to be used together with [RDO](https://www.rdoproject.org/), [Contral TripleO Heat Templates](https://github.com/Juniper/contrail-tripleo-heat-templates) and [Contrail TripleO Puppet](https://github.com/Juniper/contrail-tripleo-puppet).
```
Comment (ankitchadha) [inline README.md:13]: I see $version used within the puppet manifests. why does README say "The current implementation is limited to support Contrail v4.0"?

File: manifests/params.pp [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
 class contrail::params(
-  $analytics         = {},
-  $analyticsdatabase = {},
-  $config            = {},
-  $control           = {},
-  $database          = {},
-  $webui             = {},
-){
-
+  $analytics                  = {},
```
Comment (ankitchadha) [inline manifests/params.pp:4]: As discussed earlier, please confirm that these params would be used only for now. Before we GA, the idea is to expose all variables in user-facing tripleo-heat-templates.

File: manifests/analytics/install.pp [hunk 3]
```diff
@@ -4,23 +4,34 @@
...
+  $container_url            = undef,
+) inherits contrail::params {
 
-  package { 'wget' :
-    ensure => latest,
+  if $version < 4 {
+    package { 'python-redis' :
+      ensure => absent,
+    } ->
+    package { 'python-gevent' :
```
Comment (ankitchadha) [inline manifests/analytics/install.pp:25]: I remember there was a version dependency with python-gevent. Nagendra can probably comment on this one.

File: manifests/container/config.pp [hunk 4]
```diff
@@ -0,0 +1,71 @@
+
+# TODO: ssl_enabled - probably use another parameter, e.g. insecure
```
Comment (ankitchadha) [inline manifests/container/config.pp]: This is already done in this manifest?

File: manifests/container/config.pp [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+# TODO: to check controller_virtual_ip, glance_api_ip, neutron_metadata_ip
+# TODO: pass from upper layer: configdb_cassandra_user, configdb_cassandra_password
+# TODO: pass from upper layer: analyticsdb_cassandra_user, analyticsdb_cassandra_password
+
+define contrail::container::config (
```
Comment (ankitchadha) [inline manifests/container/config.pp]: Plan here is to be at-par with contrail-ansible with respect to exposing variables (changes will need to be made to upper layers also)

**Review by ankitchadha — APPROVED** [review]
Submitting my changes.

File: README.md (hunk 1 above)
Comment (alexey-mr) [reply README.md:13]: just because v4 is only tested for now.

File: manifests/params.pp (hunk 2 above)
Comment (alexey-mr) [reply manifests/params.pp:4]: yes, container_url and container_tag are exposed in my clone https://github.com/alexey-mr/contrail-tripleo-heat-templates/tree/stable/newton.  Let me know if there is something else to be exposed.

**Pull request closed without merging**